src/ui/public/timefilter/lib/diff_interval.js

```diff
@@ -1,14 +1,24 @@
 import _ from 'lodash';
 import { UtilsDiffTimePickerValsProvider } from 'ui/utils/diff_time_picker_vals';
 
-export function TimefilterLibDiffIntervalProvider(Private) {
+export function TimefilterLibDiffIntervalProvider(Private, $interval, $rootScope) {
   const diff = Private(UtilsDiffTimePickerValsProvider);
 
   return function (self) {
     let oldRefreshInterval = _.clone(self.refreshInterval);
+    let intervalTimeout;
 
     return function () {
       if (diff(self.refreshInterval, oldRefreshInterval)) {
+        const refreshValue = self.refreshInterval.value;
+        const refreshPause = self.refreshInterval.pause;
+
+        if (_.isNumber(refreshValue) && refreshValue > 0 && !refreshPause) {
+          intervalTimeout = $interval(() => $rootScope.$broadcast('courier:searchRefresh'), refreshValue);
```

Comment:
(Btw, I'm not familiar with Courier, so this might be wrong.)

From what I can tell, the search looper re-schedules _after_ completion, which this doesn't. This will just schedule something to happen every `refreshValue` ms. Let's say the connection is choppy, and the previous request ends up taking a long time, then a new request is scheduled which completes fast, then the old one completes. Then you can end up in a weird state.

Not sure if this is an actual problem that could end up happening here, but I've hit it a few times before when using `interval` instead of scheduling a new `timeout` on completion.

Reply:
You make a good point, and actually after looking more into this, the search looper is indeed used, by saved searches.  Only visualize listens to `courier:searchRefresh` to do the reloading, but if you have a dashboard with a bunch of saved searches, they rely on the search looper.

I'll retract this PR for now, though it's pretty awkward that our two embedded systems use such different ways of achieving this. I actually think the search looper is a better idea than listening for the global `courier:searchRefresh` signal.